packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/string_state_var.test.rs

```diff
@@ -0,0 +1,209 @@
+    let mut state_var = StringStateVar::new_from_children(String::from("")).into_state_var();
+    let mut state_var_view = state_var.create_new_read_only_view();
+
+    //////////////////////////////////////////////////
+    // Step 1: check that get the correct data queries
+    //////////////////////////////////////////////////
+    let queries = state_var.return_data_queries();
+
+    // should return a query for text children
+    assert_eq!(
+        queries,
+        vec![DataQuery::Child {
+            match_profiles: vec![ComponentProfile::Text],
+            exclude_if_prefer_profiles: vec![],
+            always_return_value: true,
+        },]
+    );
+
+    ////////////////////////////////////////////////////////////////
+    // Step 2: fulfill data query with two string children
+    ////////////////////////////////////////////////////////////////
+    let (child1_dependency, child1_var) =
+        create_state_var_dependency(String::from("Hello "), false);
+    let (child2_dependency, child2_var) = create_state_var_dependency(String::from("World"), false);
+
+    let dependencies_created_for_data_queries = vec![DependenciesCreatedForDataQuery(vec![
+        child1_dependency,
+        child2_dependency,
+    ])];
+
+    state_var.save_dependencies(&dependencies_created_for_data_queries);
+
+    ////////////////////////////////////////////////////////////////
+    // Step 3: check if get the correct calculated value
+    ////////////////////////////////////////////////////////////////
+    state_var.calculate_and_mark_fresh();
+
+    // we expect a value of "Hello World" and came_from_default to be false,
+    // since that the variable is coming from the child dependency, which we initialized to true.
+    assert_eq!(state_var.get().clone(), "Hello World");
+    assert_eq!(state_var.came_from_default(), false);
+
+    ////////////////////////////////////////////////////////////////
+    // Step 4: cannot invert to make the value "Bye World"
+    ////////////////////////////////////////////////////////////////
+
+    // on the state variable view, record that we request the value be false
+    state_var_view.queue_update(String::from("Bye World"));
+
+    let invert_result = state_var.invert(false);
+
+    assert!(invert_result.is_err());
+
+    ////////////////////////////////////////////////////////////////
+    // Step 5: change the text children to make it "Bye Earth"
+    ////////////////////////////////////////////////////////////////
+
+    child1_var.set_value(String::from("Bye "));
+    child2_var.set_value(String::from("Earth"));
+
+    state_var.calculate_and_mark_fresh();
+
+    // now the value should be "Bye Earth"
+    assert_eq!(state_var.get().clone(), "Bye Earth");
+}
+
+/// Testing the case of string state variable based on an attribute
+/// For example, if we have a `<textInput prefill/>`
+/// its `prefill` state variable would follow this pattern.
+#[test]
+fn string_state_var_with_attribute() {
```

Comment:
I don't understand what this is supposed to test. Is it that correct data queries are returned for a string state var?

Reply:
Hope it is clearer now